content-scope-scripts/content-scope-scripts-impl/src/main/java/com/duckduckgo/contentscopescripts/impl/messaging/ContentScopeScriptsJsMessaging.kt

```diff
@@ -48,13 +54,14 @@ class ContentScopeScriptsJsMessaging @Inject constructor(
 
     private lateinit var webView: WebView
     private lateinit var jsMessageCallback: JsMessageCallback
+    private var tabId: String? = null
```

Comment:
@marcosholgado I know you're working on js messageing too.
Can you take al look at this. I have very limited context but this `var tabId` smells a bit. It seems very specific/hacky to be later part of the public API

Reply:
Agreed. All the logic in the `PermissionsQueryMessage` should be moved to the BrowserTabViewModel and get there via the BrowserTabFragment. 

The idea being that your message handler just calls `jsMessageCallback.process(method)` and everything is delegated to the tab fragment and view model. 

You can have a look at https://github.com/duckduckgo/Android/pull/3870 which should be merged soon and does something similar. That PR also standardises a bit more the JsMessageCallback so all the data received in the message is passed along (which wasn't needed at first). That means all the API changes made here can be rolled back too.